File: backend/src/NoBolso.Domain/Entities/Transacao.cs
using NoBolso.Domain.Enums;

namespace NoBolso.Domain.Entities;

public class Transacao
{
    public Guid Id { get; set; }
    public string Descricao { get; set; } = string.Empty;
    public decimal Valor { get; set; }
    public DateTime Data { get; set; }
    public TipoTransacao Tipo { get; set; }

    // Chaves Estrangeiras
    public Guid UsuarioId { get; set; }
    public Guid CarteiraId { get; set; } // Adicionado
    public Guid? CategoriaTransacaoId { get; set; } // Adicionado

    // Propriedades de Navegação
    public Usuario Usuario { get; set; } = null!;
    // Adicionar navegação para Carteira e CategoriaTransacao se necessário
    public Recorrencia? Recorrencia { get; set; } // Adicionado
}

public class Recorrencia
{
    public Guid Id { get; set; }
    public string Tipo { get; set; } = string.Empty; // monthly, weekly, etc.
    public DateTime? DataFim { get; set; }
    public bool? EInfinito { get; set; }

    // Chave Estrangeira para Transacao
    public Guid TransacaoId { get; set; }
}

File: frontend/src/types/transacao.ts
export interface Transacao {
  id: string;
  descricao: string;
  valor: number;
  tipo: 'receita' | 'despesa';
  data: Date;
  carteiraId: string;
  usuarioId: string;
  categoriaId?: string; // Para categorias de investimento
  categoriaTransacaoId?: string; // Para categorias de transação normais
  recorrencia?: {
    tipo: 'none' | 'daily' | 'weekly' | 'biweekly' | 'monthly' | 'bimonthly' | 'quarterly' | 'semester' | 'yearly' | 'custom';
    repeticoes?: number;
    dataFim?: Date;
    eInfinito?: boolean;
  };
}

export interface Carteira {
  id: string;
  nome: string;
  saldo: number;
}

export interface Usuario {
  id: string;
  nome: string;
  avatar?: string;
  cor: string;
}

export interface CategoriaInvestimento {
  id: string;
  nome: string;
  objetivo: number;
  atual: number;
  cor: string;
  criadoEm: Date;
}

export interface CategoriaTransacao {
  id: string;
  nome: string;
  tipo: 'receita' | 'despesa' | 'ambos';
  cor: string;
}

export interface DadosFormularioTransacao {
  descricao: string;
  valor: number;
  tipo: 'receita' | 'despesa';
  data: Date;
  carteiraId: string;
  usuarioId: string;
  categoriaId?: string;
  categoriaTransacaoId?: string;
  recorrencia?: {
    tipo: 'none' | 'daily' | 'weekly' | 'biweekly' | 'monthly' | 'bimonthly' | 'quarterly' | 'semester' | 'yearly' | 'custom';
    repeticoes?: number;
    dataFim?: Date;
    eInfinito?: boolean;
  };
}

export interface FiltrosTransacao {
  dataInicio?: Date;
  dataFim?: Date;
  tipo?: 'receita' | 'despesa' | 'todos';
  carteiraId?: string;
  usuarioId?: string;
}

export interface DetalheDespesaRecorrente {
  id: string;
  descricao: string;
  valorMensal: number;
  dataFim?: Date;
  eInfinito?: boolean;
  totalRestante: number;
}
